pkg/scheduler/internal/cache/snapshot_test.go

```diff
@@ -229,3 +230,107 @@ func TestCreateUsedPVCSet(t *testing.T) {
 		})
 	}
 }
+
+func TestNewSnapshot(t *testing.T) {
+	testCases := []struct {
+		name                         string
+		pods                         []*v1.Pod
+		nodes                        []*v1.Node
+		expectedNumNodes             int
+		expectedPodsWithAffinity     int
+		expectedPodsWithAntiAffinity int
+	}{
+		{
+			name:  "no pods no nodes",
+			pods:  nil,
+			nodes: nil,
+		},
+		{
+			name: "single pod single node",
+			pods: []*v1.Pod{
+				st.MakePod().Name("foo").Namespace("foo").Node("node-0").Obj(),
+			},
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0"},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNumNodes: 1,
+		},
+		{
+			name: "multiple nodes, pod with affinity",
+			pods: []*v1.Pod{
+				st.MakePod().Name("foo").Namespace("ns").Node("node-0").Obj(),
+				st.MakePod().Name("bar").Namespace("ns").PodAffinity("baz", &metav1.LabelSelector{MatchLabels: map[string]string{"baz": "qux"}}, st.PodAffinityWithRequiredReq).Node("node-2").Obj(),
+			},
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0"},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-1"},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-2", Labels: map[string]string{"baz": "qux"}},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNumNodes:         3,
+			expectedPodsWithAffinity: 1,
+		},
+		{
+			name: "multiple nodes, pod with affinity, pod with anti-affinity",
+			pods: []*v1.Pod{
+				st.MakePod().Name("foo").Namespace("ns").Volume(v1.Volume{}).Node("node-1").Obj(),
+				st.MakePod().Name("bar").Namespace("ns").PodAffinity("key", &metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}}, st.PodAffinityWithRequiredReq).Node("node-0").Obj(),
+				st.MakePod().Name("baz").Namespace("ns").PodAntiAffinity("another", &metav1.LabelSelector{MatchLabels: map[string]string{"another": "label"}}, st.PodAntiAffinityWithRequiredReq).Node("node-0").Obj(),
+			},
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0", Labels: map[string]string{"key": "value"}},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-1", Labels: map[string]string{"another": "label"}},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNumNodes:             2,
+			expectedPodsWithAffinity:     1,
+			expectedPodsWithAntiAffinity: 1,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			snapshot := NewSnapshot(test.pods, test.nodes)
+			if len(test.nodes) == 0 {
```

Comment:
This is not table-driven test looks like I think, we usually put what we expected as fields of struct `testCases`. And I think we can directly remove this part of codes. We already verify the nodes number.

Reply:
My general criteria for test tables is this: 
1) do we have good coverage for what we are trying to test? in this case yes
2) will the test likely evolve to support other cases and so should be structured in a way to allow this? in this case the likely answer is no
3) is what we are trying to test simple enough that a table-based format will bloat the test? in this case I feel the answer is yes.

Given the above, I wouldn't push back and ask for table format, but that is just my opinion.

Reply:
Thanks for the constructive advices. Really make sense to me. My consideration here is:
1) The `Get()` function here is not well-designed, most of the time, we're calling it redundant, the main reason here is we're testing the `NewSnapshot()`, not the `Get()`, so I think we can verify the snapshot itself directly, e.g. `nodeInfoMap`, but considering we already tested the NumNodes, so removing them directly also seems good to me.
2) We can add a separate test like `TestSnapshot_Get()`, if we want to cover the `Get()` method, I can think of three cases:
    1. Node exists
    2. Node not exists
    3. Node exists, but is nil

So my comment is not all about the testcase style, my fault. After all, the PR definitely improves the overall code health of the project, so LGTM.